### DISCUSSION TITLE
[feature request] want support forever running task until manual stop and Prometheus monitor

### DISCUSSION THREAD

**Discussion by pengyusong** [apache/airflow#39868] [General] [answered]
### Description

1. support forever running task until manual stop
Now stream task gets more and more popular, Airflow should support forever running task, until use manual stop or timed check task status. for example, kafkaconnect start one connector, it will forever running, unless stop or fail itself.

2. support Prometheus monitor 
Prometheus should be support

### Use case/motivation

We want all data pipeline task maintain by airflow or other tools, include flink/spark streaming task, or kafkaconnect streaming task.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment, marked as answer]
There are sensors that essentially do that - if you start external task, you can put a sensor that will wait until that task runs. Prometheus is supported via statsd integration or via Open Telemetry. See our metrics - also our helm chart has prometheus statsd exporter.

BTW. Those things - if you are not sure sure if they are supported, are better for discussion, not an issue, consider starting discussion instead if you have not found answer for your needs by thoroughly reading the documentation. Also - It's a good practice to start one topic per thread. if you add more topics, very quickly such a discussion becomes a mess - thry to use the fact that there are clear thread names that allow to organize the discussions.

Converting it to discussion in case more discussion is needed.

**Reply by pengyusong** [reply]
> BTW. Those things - if you are not sure sure if they are supported, are better for discussion, not an issue, consider starting discussion instead if you have not found answer for your needs by thoroughly reading the documentation. Also - It's a good practice to start one topic per thread. if you add more topics, very quickly such a discussion becomes a mess - thry to use the fact that there are clear thread names that allow to organize the discussions.

Sorry, I got it, Thanks for that. And ... Could you direct give the links that how to custom one task, eg kafkaconnect external task and timed check status

**Reply by potiuk** [reply]
Just look for sensors in our docs
